src/main/java/com/zerobase/timate/service/UserService.java

```diff
@@ -0,0 +1,68 @@
+package com.zerobase.timate.service;
+
+import com.zerobase.timate.compoent.MailComponent;
+import com.zerobase.timate.dto.SignUpForm;
+import com.zerobase.timate.entity.User;
+import com.zerobase.timate.repository.UserRepository;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+	private final UserRepository userRepository;
+	private final MailComponent mailComponent;
+
+	@Value("${SERVER_URL}")
+	private String serverUrl;
+
+	public String signUp(SignUpForm form) {
+		Optional<User> optionalUser = userRepository.findByEmail(form.getEmail());
+		if (optionalUser.isPresent()) {
+			// TODO : custom exception
+			return "이미 가입된 회원입니다.";
+		}
```

Comment:
아래의 두 방법으로 대신해볼수 있을 것 같아요. 

```suggestion
		if (userRepository.findByEmail(form.getEmail()).isPresent()) {
			// TODO : custom exception
			return "이미 가입된 회원입니다.";
		}
```
가능하면 Optiona<User> 와 같이 Optional type을 로컬 변수로 받지 않는 것이 좋습니다

```suggestion
		if (userRepository.existByEmail(form.getEmail())) {
			// TODO : custom exception
			return "이미 가입된 회원입니다.";
		}
```
exist로 대신해볼수도 있을 것 같아요